### PR TITLE
feat: template view tags

### PR DESCRIPTION
**Pull request by csiyang**
# Description

### <samp>🤖 Generated by Copilot at e80c9eb</samp>

This pull request adds new components and icons to display the tags of a journey template in the template view. It creates a `TemplateTags` component that renders a list of `TagItem` components, each with a tag icon and name. It also adds two new icons, `Calendar4` and `SmileyNeutral`, to the `Icon` component and Storybook.

- Link to Basecamp Todo

# How should this PR be QA Tested?

Please describe the QA tests that you ran to verify your changes. Please also note any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

# Walkthrough

### <samp>🤖 Generated by Copilot at e80c9eb</samp>

*  Add two new icon components `Calendar4` and `SmileyNeutral` that render SVG paths of calendar and smiley face icons respectively, using the `createSvgIcon` utility from MUI ([link](https://github.com/JesusFilm/core/pull/1956/files?diff=unified&w=0#diff-3eca4ba97126b1942c2f2525af79d78985994912ef6ecfefbf756a831f09397dR1-R10), [link](https://github.com/JesusFilm/core/pull/1956/files?diff=unified&w=0#diff-23df947d1b31c8319f34abd36498f4d909237e0c073595338c1d3fb02c928d6cR1-R10))
*  Import the new icon components into the `Icon` component, which is a wrapper component for rendering different icons based on a name prop ([link](https://github.com/JesusFilm/core/pull/1956/files?diff=unified&w=0#diff-da0191822425b46652c3a94fd0d1226b0c63f9e931af906db352a58c777bcf59R36), [link](https://github.com/JesusFilm/core/pull/1956/files?diff=unified&w=0#diff-da0191822425b46652c3a94fd0d1226b0c63f9e931af906db352a58c777bcf59R153))
*  Add the new icon names to the `IconName` type, which is a union type of all the possible icon names that the `Icon` component can accept as a prop ([link](https://github.com/JesusFilm/core/pull/1956/files?diff=unified&w=0#diff-da0191822425b46652c3a94fd0d1226b0c63f9e931af906db352a58c777bcf59R227), [link](https://github.com/JesusFilm/core/pull/1956/files?diff=unified&w=0#diff-da0191822425b46652c3a94fd0d1226b0c63f9e931af906db352a58c777bcf59R344))
*  Add the new icon components to the `iconComponents` object, which is a mapping of icon names to icon components that the `Icon` component uses to render the correct icon based on the name prop ([link](https://github.com/JesusFilm/core/pull/1956/files?diff=unified&w=0#diff-da0191822425b46652c3a94fd0d1226b0c63f9e931af906db352a58c777bcf59R422), [link](https://github.com/JesusFilm/core/pull/1956/files?diff=unified&w=0#diff-da0191822425b46652c3a94fd0d1226b0c63f9e931af906db352a58c777bcf59R539))
*  Add the new icon names to the `iconNames` array, which is a list of all the icon names that the `Icon` component can accept as a prop, and use it to generate stories for the `Icon` component in Storybook, which is a tool for developing and testing UI components in isolation ([link](https://github.com/JesusFilm/core/pull/1956/files?diff=unified&w=0#diff-488d261fceb4cfb4cde04e83606f5653cf12470ec3d92e14cc28c7f9757bf70aR44), [link](https://github.com/JesusFilm/core/pull/1956/files?diff=unified&w=0#diff-488d261fceb4cfb4cde04e83606f5653cf12470ec3d92e14cc28c7f9757bf70aR161))
*  Add a new component `TagItem` that renders a tag icon and name based on the tag data from the journey, using a switch statement to determine the appropriate icon for the parent tag id, and using MUI Stack and Typography components for layout and styling ([link](https://github.com/JesusFilm/core/pull/1956/files?diff=unified&w=0#diff-a14c5e8e4efa05f01ee9943aad47512379b42c2c528d0d272f029eea0fc0cfc9R1-R41))
*  Export the `TagItem` component from its own index file for easier importing ([link](https://github.com/JesusFilm/core/pull/1956/files?diff=unified&w=0#diff-e02526380ee7ab5cb33699e9c8fc3832142a8b0025308a1c7db87101d6d964f6R1))
*  Add a new component `TemplateTags` that renders a horizontal scrollable list of `TagItem` components based on the tags prop from the journey, using MUI Stack and Divider components for layout and styling, and adding some custom CSS to hide the scrollbar ([link](https://github.com/JesusFilm/core/pull/1956/files?diff=unified&w=0#diff-27f60c8016f090fd3d4e18b25cf7967e2bbd58769825e8c8e8c54744a332a916R1-R39))
*  Export the `TemplateTags` component from its own index file for easier importing ([link](https://github.com/JesusFilm/core/pull/1956/files?diff=unified&w=0#diff-ac2ca8ce8b3644478d5c2a9de49ab5948cf09d1e20491b942a4831491e5e93b3R1))
*  Import the `TemplateTags` component into the `TemplateView` component, which is the main component for displaying the journey template details ([link](https://github.com/JesusFilm/core/pull/1956/files?diff=unified&w=0#diff-436738bb5e18937497677275cf375e4dbebb3183971ed6225561bed0c82bd150R17))
*  Add the `TemplateTags` component to the `TemplateView` component, passing the journey tags as a prop, and placing it below the journey title and above the journey description ([link](https://github.com/JesusFilm/core/pull/1956/files?diff=unified&w=0#diff-436738bb5e18937497677275cf375e4dbebb3183971ed6225561bed0c82bd150R45))

### CODE REVIEW

File: apps/journeys-admin/src/components/TemplateView/TemplateView.tsx [hunk 1]
```diff
@@ -41,6 +42,7 @@ export function TemplateView({ authUser }: TemplateViewProps): ReactElement {
   return (
     <Stack gap={4}>
       <TemplateViewHeader isPublisher={isPublisher} authUser={authUser} />
+      <TemplateTags tags={journey?.tags} />
```
Comment (Kneesal) [inline apps/journeys-admin/src/components/TemplateView/TemplateView.tsx:45]: just double check the spacing for desktop and mobile :)

File: apps/journeys-admin/src/components/TemplateView/TemplateTags/getParentTagsWithIcon/getParentTagsWithIcon.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+      },
+      {
+        __typename: 'Tag',
+        id: 'tag6.id',
+        parentId: 'otherTag.id',
+        service: null,
+        name: [
+          {
+            __typename: 'Translation',
+            value: 'tag6',
```
Comment (Kneesal) [inline apps/journeys-admin/src/components/TemplateView/TemplateTags/getParentTagsWithIcon/getParentTagsWithIcon.spec.tsx:107]: value: "Collection"

File: apps/journeys-admin/src/components/TemplateView/TemplateTags/getParentTagsWithIcon/getParentTagsWithIcon.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+              __typename: 'Language',
+              id: '529'
+            }
+          }
+        ]
+      },
+      {
+        __typename: 'Tag',
+        id: 'tag6.id',
+        parentId: 'otherTag.id',
```
Comment (Kneesal) [inline apps/journeys-admin/src/components/TemplateView/TemplateTags/getParentTagsWithIcon/getParentTagsWithIcon.spec.tsx:102]: parentId: null

**Review by Kneesal — DISMISSED** [review]

File: apps/journeys-admin/src/components/TemplateView/TemplateTags/getParentTagsWithIcon/getParentTagsWithIcon.tsx [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+  id: string
+  name: string
+  icon: ReactNode
+}
+
+export function getParentTagsWithIcon(parentTags?: Tag[]): ParentTagWithIcon[] {
+  const parentTagsWithIcons: ParentTagWithIcon[] = []
+
+  parentTags?.forEach((tag) => {
+    const name = tag.name.find((name) => name.language.id === '529')?.value
```
Comment (tataihono) [inline apps/journeys-admin/src/components/TemplateView/TemplateTags/getParentTagsWithIcon/getParentTagsWithIcon.tsx:22]:
```suggestion
    const ENGLISH_LANGUAGE_ID = '529'
    const name = tag.name.find((name) => name.language.id === ENGLISH_LANGUAGE_ID)?.value
```

https://eslint.org/docs/latest/rules/no-magic-numbers i really like this rule. perhaps we should enforce it. The problem here is that you just have to magically know that 529 is english.

**Review by tataihono — CHANGES_REQUESTED** [review]

**Review by tataihono — CHANGES_REQUESTED** [review]

File: apps/journeys-admin/src/components/TemplateView/TemplateTags/TemplateTags.tsx [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+            {tagItems.map(({ id, name, icon }, index) => (
+              <SwiperSlide key={id} style={{ width: 'fit-content', zIndex: 2 }}>
+                <TagItem
+                  key={id}
+                  name={name}
+                  icon={icon}
+                  showDivider={index < tagItems.length - 1}
+                />
+              </SwiperSlide>
+            ))}
```
Comment (tataihono) [inline apps/journeys-admin/src/components/TemplateView/TemplateTags/TemplateTags.tsx:52]:
```suggestion
            {sortedTags.map(({ id, name, parentId }, index) => (
              <SwiperSlide key={id} style={{ width: 'fit-content', zIndex: 2 }}>
                <TagItem
                  key={id}
                  name={name.find(({ primary }) => primary).value}
                  icon={<ParentTagIcon
        name={
          parentTags?.find((parentTag) => parentTag.id === tag.parentId)
            ?.name[0].value
        }
      />}
                  showDivider={index < sortedTags.length - 1}
                />
              </SwiperSlide>
            ))}
```

File: apps/journeys-admin/src/components/TemplateView/TemplateTags/TemplateTags.tsx [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+          </SwiperWrapper>
+        )
+      ) : (
+        <SwiperWrapper>
+          {[0, 1, 2].map((item, index, array) => (
+            <SwiperSlide key={item} style={{ width: '128px', zIndex: 2 }}>
```
Comment (tataihono) [inline apps/journeys-admin/src/components/TemplateView/TemplateTags/TemplateTags.tsx:58]: why does width have to be specified manually here?

**Review by tataihono — DISMISSED** [review]

File: apps/journeys-admin/src/components/TemplateView/TemplateTags/TemplateTags.tsx (hunk 5 above)
Comment (csiyang) [reply apps/journeys-admin/src/components/TemplateView/TemplateTags/TemplateTags.tsx:52]: Still need to get some from the parent tags list

File: apps/journeys-admin/src/components/TemplateView/TemplateTags/TemplateTags.tsx (hunk 6 above)
Comment (csiyang) [reply apps/journeys-admin/src/components/TemplateView/TemplateTags/TemplateTags.tsx:58]: Redundant now with the new swipper changes

**Review by storyworks — DISMISSED** [review]

**Review by storyworks — APPROVED** [review]

**Pull request merged**